addons/mass_mailing/models/mailing.py

```diff
@@ -54,6 +55,14 @@ def default_get(self, fields_list):
                 mailing_list = self.env['mailing.list'].search([], limit=2)
                 if len(mailing_list) == 1:
                     vals['contact_list_ids'] = [(6, 0, [mailing_list.id])]
+
+        # When mass mailing in enabled for other apps(for example, we redirect user to
+        # mass mailing form view from contacts' list view once mass mailing is installed),
+        # set the proper model/domain instead of the default ones.
+        if context.get('active_model') and 'mailing_model_id' in fields_list:
```

Comment:
Can't you instead set the default_model_id and default_mailing_domain directly in action ? active_ids is normally usable in evaluation context (and ref for model should work, otherwise a default_mailing_model_name to convert back to a real model in default_get). Like (did not test, just to try it out):
```
<field name="context">{
    'default_mailing_domain': [('id', 'in', active_ids)],
    'default_mailign_model_id': ref('base.model_res_partner'),
    'default_reply_to': '',
    'default_reply_to_mode': 'new',
}</field>
```

Reply:
ello @tde-banana-odoo ,

We can't use ref when storing context as a string . This can be achieved with `eval` attribute on the field, but then it will be evaluated while loading the record from xml, so we'll face trouble with `active_ids`. So, currently I am passing

```
<field name="context">{
            'default_mailing_domain': [('id', 'in', active_ids)],
            'default_mailing_model_name': 'res.partner',
            'default_reply_to': '',
            'default_reply_to_mode': 'new',
        }</field>
```
and then retrieving id of the model from it's name in default_get as you suggested.

Reply:
Indeed, we will have to use default_model_name and convert it back, that's ok :) . Thanks for checking !